specs/dispute-game-interface.md

```diff
@@ -24,53 +25,188 @@ dispute games to exist. If multiple dispute games run in production, it gives
+
+    /// @notice `gameImpls` is a mapping that maps `GameType`s to their respective `IDisputeGame` implementations.
+    /// @param gameType The type of the dispute game.
+    /// @return _impl The address of the implementation of the game type. Will be cloned on creation.
+    function gameImpls(GameType gameType) public view returns (IDisputeGame _impl);
+
+    /// @notice The owner of the contract.
+    /// @dev Owner Permissions:
+    ///      - Update the implementation contracts for a given game type.
+    /// @return _owner The owner of the contract.
+    function owner() public view returns (address _owner);
+
+    /// @notice Creates a new DisputeGame proxy contract.
+    /// @param gameType The type of the DisputeGame - used to decide the proxy implementation
+    /// @param rootClaim The root claim of the DisputeGame.
+    /// @param extraData Any extra data that should be provided to the created dispute game.
+    function create(GameType gameType, Claim rootClaim, bytes calldata extraData) external returns (IDisputeGame proxy);
+
+    /// @notice Sets the implementation contract for a specific `GameType`
+    /// @param gameType The type of the DisputeGame
+    /// @param impl The implementation contract for the given `GameType`
+    function setImplementation(GameType gameType, IDisputeGame impl) external;
 }
+```
+
+## `DisputeGame` Interface
+
+The dispute game interface should be generic enough to allow it to work with any
+proof system. This means that it should work fault proofs, validity proofs,
+an attestation based proof system, or any other source of truth that adheres to
+the interface.
+
+Clones of the `IDisputeGame`'s `initialize` functions will be called by the `DisputeGameFactory` upon creation.
+
+```solidity
+////////////////////////////////////////////////////////////////
+//                    GENERIC DISPUTE GAME                    //
+////////////////////////////////////////////////////////////////
 
 /// @title IDisputeGame
 /// @notice The generic interface for a DisputeGame contract.
 interface IDisputeGame {
     /// @notice Initializes the DisputeGame contract.
-    /// @dev It is recommended that the implementations of this interface only allow this function to be called once.
+    /// @custom:invariant The `initialize` function may only be called once.
     function initialize() external;
 
     /// @notice Returns the semantic version of the DisputeGame contract
     function version() external pure returns (string memory _version);
 
+    /// @notice Returns the timestamp that the DisputeGame contract was created at.
+    function createdAt() external pure returns (Timestamp _createdAt);
+
+    /// @notice Returns the current status of the game.
+    function status() external view returns (GameStatus _status);
+
+    /// @notice `clones-with-immutable-args` argument #1
     /// @return _gameType The type of proof system being used.
```

Comment:
Minor nit, but this seems like an internal detail that should be behind a dev attribute.

```suggestion
    /// @return _gameType The type of proof system being used.
    /// @dev `clones-with-immutable-args` argument #1
```